### PR TITLE
test doctests & improve deprecated decorator

### PR DESCRIPTION
**Pull request by kernc**

### CODE REVIEW

File: Orange/evaluation/scoring.py [hunk 1]
```diff
@@ -1,3 +1,15 @@
+"""
+Methods of scoring prediction results (CA, AUC, ...).
```
Comment (astaric) [inline Orange/evaluation/scoring.py]: for?

**Comment by lanzagar** [comment]
Before this change our unittests ended with:

```
test_outputs (Orange.widgets.tests.test_widgets_outputs.TestWidgetOutputs) ... /usr/lib/python3.5/site-packages/sklearn/lda.py:4: DeprecationWarning: lda.LDA has been moved to discriminant_analysis.LinearDiscriminantAnalysis in 0.17 and will be removed in 0.19
  "in 0.17 and will be removed in 0.19", DeprecationWarning)
/home/lan/.local/lib/python3.5/site-packages/nltk-3.1-py3.5.egg/nltk/decorators.py:59: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() instead
  regargs, varargs, varkwargs, defaults = inspect.getargspec(func)
/usr/lib/python3.5/site-packages/numpy/lib/utils.py:99: DeprecationWarning: `scipy.sparse.sparsetools` is deprecated!
scipy.sparse.sparsetools is a private module for scipy.sparse, and should not be used.
  warnings.warn(depdoc, DeprecationWarning)
/home/lan/.local/lib/python3.5/site-packages/orangecontrib/text/nyt.py:13: DeprecationWarning: 'Orange.canvas.utils.environ' is deprecated please do not import it
  from Orange.canvas.utils import environ
ok
```

now the last DeprecationWarning does not occur:

```
test_outputs (Orange.widgets.tests.test_widgets_outputs.TestWidgetOutputs) ... ok
```

Any idea what is going on @kernc @ales-erjavec ?

**Comment by kernc** [comment]
First guess is warnings filter (re)set somewhere. But I can't seem to find it. :smiley: 
DeprecationWarnings are [off by default](https://docs.python.org/3/library/warnings.html#default-warning-filters).
```text
314a315,332
> deprecated (Orange.util)
> Doctest: Orange.util.deprecated ... ok
> Remove (Orange.preprocess.remove)
> Doctest: Orange.preprocess.remove.Remove ... ok
> Normalize (Orange.preprocess.preprocess)
> Doctest: Orange.preprocess.preprocess.Normalize ... ok
> Randomize (Orange.preprocess.preprocess)
> Doctest: Orange.preprocess.preprocess.Randomize ... ok
> WrapperMeta (Orange.misc.wrapper_meta)
> Doctest: Orange.misc.wrapper_meta.WrapperMeta ... ok
> scoring (Orange.evaluation)
> Doctest: Orange.evaluation.scoring ... ok
> F1 (Orange.evaluation.scoring)
> Doctest: Orange.evaluation.scoring.F1 ... ok
> LogLoss (Orange.evaluation.scoring)
> Doctest: Orange.evaluation.scoring.LogLoss ... ok
> _rxc_ix (Orange.data.table)
> Doctest: Orange.data.table._rxc_ix ... ok
361,363c379
< test_auc_on_multiclass_data_returns_1d_array (test_evaluation_scoring.Scoring_AUC_Test) ... /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/site-packages/numpy/lib/function_base.py:1175: DeprecationWarning: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
<   return a[slice1]-a[slice2]
< ok
---
> test_auc_on_multiclass_data_returns_1d_array (test_evaluation_scoring.Scoring_AUC_Test) ... ok
387,389c403
< test_too_many_folds (test_evaluation_testing.CrossValidationTestCase) ... /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/site-packages/sklearn/cross_validation.py:516: Warning: The least populated class in y has only 50 members, which is too few. The minimum number of labels for any class cannot be less than n_folds=75.
<   % (min_labels, self.n_folds)), Warning)
< ok
---
> test_too_many_folds (test_evaluation_testing.CrossValidationTestCase) ... ok
528,544c542,544
< test_missing_class (test_regression.MissingValuesTest) ... /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/site-packages/sklearn/linear_model/coordinate_descent.py:466: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations
<   ConvergenceWarning)
< Traceback (most recent call last):
<   File "/home/jk/PycharmProjects/orange3/Orange/tests/test_regression.py", line 61, in test_missing_class
<     learner = learner()
< TypeError: __init__() missing 1 required positional argument: 'learner'
< ok
< test_adequacy_all_learners (test_regression.RegressionLearnersTest) ... Traceback (most recent call last):
<   File "/home/jk/PycharmProjects/orange3/Orange/tests/test_regression.py", line 38, in test_adequacy_all_learners
<     learner = learner()
< TypeError: __init__() missing 1 required positional argument: 'learner'
< ok
< test_adequacy_all_learners_multiclass (test_regression.RegressionLearnersTest) ... Traceback (most recent call last):
<   File "/home/jk/PycharmProjects/orange3/Orange/tests/test_regression.py", line 48, in test_adequacy_all_learners_multiclass
<     learner = learner()
< TypeError: __init__() missing 1 required positional argument: 'learner'
< ok
---
> test_missing_class (test_regression.MissingValuesTest) ... ok
> test_adequacy_all_learners (test_regression.RegressionLearnersTest) ... ok
> test_adequacy_all_learners_multiclass (test_regression.RegressionLearnersTest) ... ok
729,731c729
< test_navigate_tree (test_tree.SklearnTreeTest) ... /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/site-packages/sklearn/utils/validation.py:386: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and willraise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.
<   DeprecationWarning)
< ok
---
> test_navigate_tree (test_tree.SklearnTreeTest) ... ok
839,852c837
< test_outputs (Orange.widgets.tests.test_widgets_outputs.TestWidgetOutputs) ... /home/jk/PycharmProjects/orange3/Orange/widgets/settings.py:246: UserWarning: Could not read defaults for widget <class 'Orange.widgets.visualize.owsieve.OWSieveDiagram'>.
< The following error occurred:
< 
< Ran out of input
<   "The following error occurred:\n\n%s" % e)
< /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/site-packages/orangecontrib/bio/kegg/conf.py:41: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
<   parser.readfp(StringIO(default), "default")
< /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/importlib/_bootstrap.py:321: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility
<   return f(*args, **kwds)
< /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/site-packages/numpy/lib/utils.py:99: DeprecationWarning: `scipy.sparse.sparsetools` is deprecated!
< scipy.sparse.sparsetools is a private module for scipy.sparse, and should not be used.
<   warnings.warn(depdoc, DeprecationWarning)
< /home/jk/PycharmProjects/venvs/orange3/lib/python3.4/importlib/_bootstrap.py:321: DeprecationWarning: 'Orange.canvas.utils.environ' is deprecated please do not import it
<   return f(*args, **kwds)
---
> test_outputs (Orange.widgets.tests.test_widgets_outputs.TestWidgetOutputs) ... MLPLearner cannot be used with default parameters
856c841
< Ran 766 tests in 18.501s
---
> Ran 775 tests in 18.081s
859d843
< MLPLearner cannot be used with default parameters
```

**Comment by kernc** [comment]
It wasn't simple filter. The `deprecated` doctest assigned `sys.stdout` to `sys.stderr` in order to catch those warnings ... :melon:

**Comment by kernc** [comment]
No changes made.

**Pull request merged**